include/deal.II/lac/precondition.h

```diff
@@ -476,6 +476,24 @@ class PreconditionJacobi : public PreconditionRelaxation<MatrixType>
 {
 public:
   /**
+   * AdditionalData to store relaxation parameter.
+   */
+  struct AdditionalData : public PreconditionRelaxation<MatrixType>::AdditionalData
```

Comment:
Here and in the other places, there is no actual need to declare this `struct` -- it is entirely equivalent to the one in `PreconditionRelaxation`. Because `PreconditionRelaxation` is a base class, the structure is actually already available, though one may use through `typename` to get at it. But rather than declare this class, declare a constructor, and simply pass the argument down to the base class, why not just say
```
  using PreconditionRelaxation<MatrixType>::AdditionalData;
```
here to make the `struct` available as a member of the current class as well?